### PR TITLE
Add Program Manager governance docs

### PR DESCRIPTION
**Pull request by LappleApple**
This PR includes governance documentation related to the SIG Program Manager role and is a response to [#5563](https://github.com/kubernetes/community/pull/5563).

/assign @saschagrunert @LappleApple @nikhita
cc: @kubernetes/sig-release-leads @kubernetes/sig-contributor-experience-leads @kubernetes/steering-committee

/sig release
/sig contributor-experience

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by SergeyKanzhelev — COMMENTED** [review]
a few minor suggestions, lgtm otherwise

**Comment by LappleApple** [comment]
Hi @kubernetes/steering-committee folks, given the roles issue/discussion I'm wondering if anyone has reviewed my work here and/or has any thoughts or questions about it.

**Comment by LappleApple** [comment]
Sorry, want to be super-clear: the role description writeup itself, independent of the broader discussion happening around Dims' issue. Thanks!

File: committee-steering/governance/sig-governance.md [hunk 1]
```diff
@@ -140,6 +141,17 @@ curation from other SIG participants
   - Number: 2-3
   - Membership tracked in [sigs.yaml]
 
+### Program Manager
+
+- *Optional Role*: SIG Program Managers
+  -  Coordinate the success of their SIGs’ delivery workflow and related practices
+  -  Coordinate cross-SIG coordination
+  -  Set up, launch, and continuously optimize processes
+  -  Drive prioritization, delegation, communication, risk mitigation efforts
```
Comment (derekwaynecarr) [inline committee-steering/governance/sig-governance.md:150]: I have some concerns that this phrasing, i.e. "drive prioritization", may have a risk when applied broadly across all SIGs.

It should clarify that if a role like this was adopted, the individuals in this role do not control resource allocation within a SIG.  At some level, driving prioritization requires mutual consensus between the individual proposing a change and the individual reviewing it both in time and support and I worry this language could be easily confused as delegating greater powers than intended.

**Comment by cblecker** [comment]
Having discussed this PR, the Steering Committee has decided that we should not add a project-wide Program Manager role at this time. We are currently working on https://github.com/kubernetes/community/pull/5736 which will better define how SIGs can create roles and where they should be documented.

Thank you for starting this discussion. While this PR didn't result in governance changes directly, the resulting discussion has been a healthy one for our community to have.

/close

**Pull request closed without merging**